src/utils/wallet.ts

```diff
@@ -158,7 +158,7 @@ const wallet = {
     } catch (error) {
       throw new XPubError(error.message);
     }
-    const key = xpub.derive(index);
+    const key = xpub.deriveChild(index);
```

Comment:
Same comment about the `deriveChild`. If we decide to continue using it I think we should have a comment explaining why we can safely use it with xpub and must use `deriveNonCompliantChild` with xpriv.

Reply:
Is there a `deriveNonCompliantChild()` for pubkeys? If yes, I feel we should use it as well.

Reply:
There isn't, the tests on commit 2abe487 failed because of this, i chose to use deriveChild but derive and deriveChild are interchangeable (derive calls deriveChild internally, [here](https://github.com/bitpay/bitcore/blob/v8.25.10/packages/bitcore-lib/lib/hdpublickey.js#L114)). The only difference is that maybe derive can be fully deprecated on a future version.

Reply:
As we talked in the daily meeting yesterday, I think you should have a comment in all `deriveChild` uses to explain that `deriveChild` must be used ONLY for xpubs. Xprivs must always use `deriveNonCompliantChild` for now and to make sure in the future nobody will 'copy' xpub derivation method and use to derive xpriv as well